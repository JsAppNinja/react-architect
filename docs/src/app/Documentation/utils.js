export const splitOnCaps = s => s.split(/(?=[A-Z1-9])/);

function getString(s) {
  return !(s instanceof Array) ? splitOnCaps(s) : s;
}

export const toDashedName = (s) => getString(s).map(s => s.toLowerCase()).join('-');
export const toTitle = (s) => getString(s).join(' ');
