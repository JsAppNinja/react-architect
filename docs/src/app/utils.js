import React from 'react';
export const githubHref = 'https://github.com/mlaursen/react-md';

const lorems = [
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus at placerat nulla. Morbi eu erat diam. Donec malesuada mauris in neque efficitur luctus. Vestibulum porttitor elit velit, pellentesque imperdiet tellus maximus in. Phasellus consectetur lacus nunc, quis efficitur felis feugiat ac. Vivamus non erat laoreet, volutpat tellus sit amet, porttitor orci. Sed scelerisque id arcu at ornare. Suspendisse potenti. Phasellus eu posuere mi. Fusce venenatis odio dui, ac accumsan lacus dapibus et. Praesent euismod nulla at porta elementum. Nunc ullamcorper velit aliquam velit faucibus tempus.',
  'Duis ligula elit, elementum a nisl a, venenatis fermentum libero. Cras bibendum dignissim nisl, sed sodales tellus scelerisque sed. Sed at enim semper, imperdiet elit et, venenatis magna. Morbi vestibulum purus in ante placerat viverra. Nullam molestie arcu ut dolor rutrum varius. Sed vehicula laoreet volutpat. Quisque feugiat, nisl in sollicitudin consectetur, mi metus aliquet urna, ac sagittis nulla nulla in diam. Fusce id enim auctor ante fringilla auctor in et quam.',
  'Donec pulvinar ligula augue, et tincidunt velit bibendum non. Aenean id metus nunc. Donec non feugiat nisl. In accumsan ligula purus, ut vulputate elit tempus sit amet. Nullam volutpat semper lectus. Nunc a libero tellus. Duis posuere nisi diam, nec euismod risus interdum sit amet. Pellentesque at lacus bibendum, cursus ante non, maximus arcu. Morbi id leo nulla. Integer quis feugiat quam. Etiam auctor pharetra vestibulum.',
  'Nulla sodales metus metus. Sed a metus quis ipsum sagittis placerat. Nunc sit amet molestie purus, nec rutrum quam. Praesent elementum risus eget mattis condimentum. Morbi dignissim, dolor et finibus sollicitudin, nulla enim molestie est, eget tincidunt magna libero et dolor. Fusce tempor sapien et urna sodales, at laoreet nunc egestas. Aenean eget arcu in felis ultrices vehicula. Maecenas interdum viverra sem vitae sodales. Duis tincidunt varius nunc sed viverra. Duis sit amet semper ex, ultrices aliquet mi. Donec a congue sem, ac facilisis ex.',
  'Proin venenatis tortor et ex viverra, ac facilisis mi ullamcorper. Aenean eget est bibendum, fermentum enim eget, semper orci. Nunc magna mauris, fermentum sit amet diam ut, semper malesuada justo. Fusce faucibus tellus eget arcu imperdiet, aliquet placerat massa molestie. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Maecenas gravida quam id ultricies accumsan. Sed porta accumsan libero sed venenatis. Morbi scelerisque, ex in gravida porttitor, ipsum erat sodales urna, mattis mattis nibh ligula quis tellus. Pellentesque consequat et metus scelerisque dapibus. Praesent at maximus ligula. Phasellus vel orci tortor.',
  'Mauris posuere nisl sit amet justo condimentum volutpat. Aliquam venenatis rhoncus est, tristique semper tortor iaculis et. Donec pretium metus nec sodales mollis. Ut vel ipsum sapien. Fusce sit amet libero efficitur orci aliquet scelerisque a ut arcu. In hac habitasse platea dictumst. Sed turpis libero, dictum nec imperdiet nec, maximus eget urna. Sed vel lorem sit amet nisi vulputate mattis. In in odio erat. Aenean euismod lorem nec viverra auctor. Mauris id ipsum maximus, tempus dui nec, mollis felis. Etiam sodales metus sit amet lacus aliquet tristique. Nulla in lacus massa. Morbi quis lorem vitae dolor maximus luctus vel vel eros. Phasellus posuere in purus eget convallis.',
  'Aliquam mauris justo, ornare quis quam fermentum, mollis volutpat dolor. Donec sagittis velit non nisl porttitor, id lacinia tortor vulputate. Suspendisse posuere at felis eu fermentum. Donec at vehicula leo, id sollicitudin ipsum. Donec dapibus sodales erat at malesuada. Mauris sollicitudin velit sed arcu molestie congue. Suspendisse ac arcu enim. Sed ac urna id mi rutrum ornare sit amet in nisl. Phasellus at tempus magna. Nulla facilisi. Mauris pharetra finibus mi ut facilisis. Duis mattis odio id quam vestibulum, id consequat ex fringilla. Aenean sollicitudin accumsan sollicitudin. Duis euismod maximus tempor. Phasellus mi enim, bibendum at nisi vel, cursus venenatis sem. Praesent sagittis, nunc at rhoncus maximus, ligula turpis imperdiet eros, pharetra tincidunt tortor magna id sem.',
  'Sed eleifend sapien condimentum iaculis cursus. In vel pretium turpis. Duis molestie euismod turpis id consectetur. Nunc pharetra sapien libero, a volutpat erat finibus eget. Duis euismod eleifend felis consequat fringilla. Sed varius diam ligula, quis rutrum ex mollis non. Phasellus ac malesuada nibh. Praesent sit amet auctor urna. Quisque eget magna id ex dapibus vulputate eget vel magna.',
  'Ut viverra tempor placerat. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras neque diam, auctor eget sapien et, tincidunt blandit turpis. Nam non volutpat velit. Aliquam iaculis euismod nisl non facilisis. Ut blandit dignissim turpis in finibus. Cras neque augue, tempus non neque eu, ornare ultrices sem. Nullam ultricies mi quis semper pellentesque. Mauris mauris lacus, ullamcorper a iaculis quis, tempus id tellus. Phasellus in orci orci. Fusce consequat suscipit ipsum ac bibendum. Maecenas et leo eu leo iaculis varius. Nam consectetur laoreet nisl blandit condimentum.',
  'Ut dictum, sapien vel malesuada posuere, tortor tellus blandit risus, eget hendrerit quam diam eu nibh. Aliquam erat volutpat. Ut rhoncus augue ac nunc porttitor hendrerit. Pellentesque enim risus, auctor quis turpis eget, pulvinar malesuada urna. Nullam tortor mi, elementum id sodales eget, aliquam et nulla. Pellentesque efficitur mi eros, in rutrum metus ultrices vel. Proin vitae diam sit amet nunc pretium malesuada. Fusce at nisi scelerisque, ultrices tortor nec, finibus velit. Donec molestie augue eget leo rutrum, nec ornare arcu gravida. Quisque tincidunt risus elit, ut condimentum eros egestas congue.',
];
export function loremIpsum(paras = 3) {
  let ipsums = [];
  for(let i = 0; i < paras; i++) {
    ipsums.push((
      <p key={i}>{lorems[Math.floor(Math.random() * lorems.length)]}</p>
    ));
  }
  return ipsums;
}

export const numstr = ['one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine', 'ten'];

export const randomImage = ({ width, height, time } = {}) => {
  width = !width ? 200 : width;
  height = !height ? 200 : height;
  time = typeof time === 'undefined' ? Date.now() : time;
  return `https://unsplash.it/${width}/${height}/?random&time=${time}`;
};

export const hostPrefix = ''; // change to '/react-md' for production
export const imgPrefix = hostPrefix + '/imgs';
