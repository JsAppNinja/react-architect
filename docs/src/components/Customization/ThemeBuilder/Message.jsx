import React from 'react';
import { bem } from 'react-md';

const Message = () => (
  <p className={bem('theme-configuration', 'message')}>
    <i>
      When the save for future visits checkbox is checked, cookies will be created to store your
      theme for later visits and other pages. If you do not check this checkbox, the default theme
      will be applied when you leave this page.
    </i>
  </p>
);

export default Message;
