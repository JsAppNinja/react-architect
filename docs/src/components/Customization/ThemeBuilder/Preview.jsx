import React, { PureComponent } from 'react';
import { Button, Grid, Cell, Toolbar, Drawer, bem } from 'react-md';

const NAV_ITEMS = [
  { primaryText: 'Woop woop' },
  { primaryText: 'That\'s the sound' },
  { primaryText: 'Of the Police' },
];

const idPrefix = 'theme-builder-preview';

export default class Preview extends PureComponent {
  state = { visible: false };

  toggleDrawer = () => {
    this.setState({ visible: !this.state.visible });
  };

  handleVisibilityChange = (visible) => {
    this.setState({ visible });
  };

  render() {
    return (
      <section className="md-background--card theme-preview" ref={(container) => { this._container = container; }}>
        <Toolbar
          id={`${idPrefix}-toolbar`}
          nav={<Button id={`${idPrefix}-drawer-toggle`} icon onClick={this.toggleDrawer}>menu</Button>}
          title="Theme Preview"
          colored
          fixed
        />
        <Grid className="md-toolbar-relative">
          <Cell component="h2" size={12} className="md-display-1">Look at this</Cell>
          <Button id={`${idPrefix}-btn`} primary raised className={bem('theme-preview', 'btn')}>
            Button
          </Button>
        </Grid>
        <Drawer
          id={`${idPrefix}-drawer`}
          renderNode={this._container}
          header={<Toolbar title="Theme Preview" id={`${idPrefix}-drawer-toolbar`} />}
          navClassName="md-toolbar-relative"
          navItems={NAV_ITEMS}
          visible={this.state.visible}
          onVisibilityChange={this.handleVisibilityChange}
          overlay
          type={Drawer.DrawerTypes.TEMPORARY}
        />
        <Button id={`${idPrefix}-floating-btn`} floating secondary fixed>email</Button>
      </section>
    );
  }
}
