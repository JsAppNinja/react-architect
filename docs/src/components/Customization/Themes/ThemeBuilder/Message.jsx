import React from 'react';

const Message = () => (
  <i className="md-text theme-configuration__message">
    When the save for future visits checkbox is checked, cookies will be created to store your
    theme for later visits and other pages. If you do not check this checkbox, the default theme
    will be applied when you leave this page.
  </i>
);

export default Message;
