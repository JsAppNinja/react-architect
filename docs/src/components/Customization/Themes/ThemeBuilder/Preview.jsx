import React, { PureComponent } from 'react';
import Button from 'react-md/lib/Buttons/Button';
import Toolbar from 'react-md/lib/Toolbars';
import Drawer from 'react-md/lib/Drawers';

const NAV_ITEMS = [
  { primaryText: 'Woop woop' },
  { primaryText: 'That\'s the sound' },
  { primaryText: 'Of the Police' },
];

export default class Preview extends PureComponent {
  state = { visible: false };

  toggleDrawer = () => {
    this.setState({ visible: !this.state.visible });
  };

  handleVisibilityChange = (visible) => {
    this.setState({ visible });
  };

  render() {
    return (
      <section className="md-background--card theme-preview" ref={(container) => { this._container = container; }}>
        <Toolbar
          nav={<Button icon onClick={this.toggleDrawer}>menu</Button>}
          title="Theme Preview"
          colored
          fixed
        />
        <div className="md-grid md-toolbar-relative">
          <h2 className="md-display-1 md-cell md-cell--12">Look at this</h2>
          <Button primary raised className="theme-preview__btn">
            Button
          </Button>
        </div>
        <Drawer
          renderNode={this._container}
          header={<Toolbar title="Theme Preview" />}
          navClassName="md-toolbar-relative"
          navItems={NAV_ITEMS}
          visible={this.state.visible}
          onVisibilityChange={this.handleVisibilityChange}
          overlay
          type={Drawer.DrawerTypes.TEMPORARY}
        />
        <Button floating secondary fixed>email</Button>
      </section>
    );
  }
}
