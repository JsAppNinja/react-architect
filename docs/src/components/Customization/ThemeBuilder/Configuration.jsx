import React from 'react';
import PropTypes from 'prop-types';
import { SelectField, SelectionControl } from 'react-md';

import { SECONDARY_HUES, PRIMARY, SECONDARY, HUE, LIGHT } from 'constants/colors';
import Message from './Message';

const NAME = 'theme-configuration';

const Configuration = ({
  primary,
  secondary,
  hue,
  light,
  saved,
  saveDisabled,
  onChange,
  onSelectChange,
  filteredPrimaries,
  filteredSecondaries,
}) => (
  <form
    id="theme-configuration-form"
    name={NAME}
    className="md-cell md-cell--8 md-cell--6-desktop"
    onChange={onChange}
  >
    <h3 className="md-title">Theme Configuration</h3>
    <SelectionControl
      id={LIGHT}
      type="checkbox"
      name={NAME}
      defaultChecked={light}
      label="Light theme"
    />
    <SelectionControl
      id="save-theme"
      type="checkbox"
      name={NAME}
      defaultChecked={saved}
      label="Save for future visits"
      disabled={saveDisabled}
    />
    <Message />
    <SelectField
      id={PRIMARY}
      label="Primary color"
      value={primary}
      menuItems={filteredPrimaries}
      className="md-cell"
      onChange={onSelectChange}
      simplifiedMenu={false}
    />
    <SelectField
      id={SECONDARY}
      label="Secondary color"
      value={secondary}
      menuItems={filteredSecondaries}
      className="md-cell"
      onChange={onSelectChange}
      simplifiedMenu={false}
    />
    <SelectField
      id={HUE}
      label="Secondary hue"
      value={hue}
      menuItems={SECONDARY_HUES}
      className="md-cell"
      onChange={onSelectChange}
      simplifiedMenu={false}
    />
  </form>
);

Configuration.propTypes = {
  primary: PropTypes.string.isRequired,
  secondary: PropTypes.string.isRequired,
  hue: PropTypes.number.isRequired,
  light: PropTypes.bool.isRequired,
  saved: PropTypes.bool.isRequired,
  saveDisabled: PropTypes.bool.isRequired,
  filteredPrimaries: PropTypes.arrayOf(PropTypes.string).isRequired,
  filteredSecondaries: PropTypes.arrayOf(PropTypes.string).isRequired,
  onChange: PropTypes.func.isRequired,
  onSelectChange: PropTypes.func.isRequired,
};

export default Configuration;
