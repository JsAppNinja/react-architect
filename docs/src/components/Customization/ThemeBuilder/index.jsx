import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connectAdvanced } from 'react-redux';
import shallowEqual from 'shallowequal';
import Helmet from 'react-helmet';

import { updateTheme, clearTheme } from 'state/theme';
import Markdown from 'components/Markdown';

import './_styles.scss';
import Preview from './Preview';
import Configuration from './Configuration';

const ABOUT_THEME_BUILDER = `
### Custom CSS Theme Builder

Select a primary color, a secondary color, the secondary color's hue, and optionally toggle the light theme
to view a specific theme. When you have selected colors you like, either reference [Using with Sass](#using-with-sass)
or [pre-compiled themes](#pre-compiled-themes). Not all themes will already be compiled and hosted on \`unpkg\`.
`;

export class PureThemeBuilder extends PureComponent {
  static propTypes = {
    primary: PropTypes.string.isRequired,
    secondary: PropTypes.string.isRequired,
    hue: PropTypes.number.isRequired,
    light: PropTypes.bool.isRequired,
    saved: PropTypes.bool.isRequired,
    saveDisabled: PropTypes.bool.isRequired,
    href: PropTypes.string,
    filteredPrimaries: PropTypes.arrayOf(PropTypes.string).isRequired,
    filteredSecondaries: PropTypes.arrayOf(PropTypes.string).isRequired,
    clearTheme: PropTypes.func.isRequired,
    updateTheme: PropTypes.func.isRequired,
  };

  constructor(props) {
    super();

    this.state = this.getNextState(props);
  }

  componentWillReceiveProps(nextProps) {
    const { primary, secondary, hue } = this.props;
    if (primary !== nextProps.primary || secondary !== nextProps.secondary || hue !== nextProps.hue) {
      this.setState(this.getNextState(nextProps));
    }
  }

  componentWillUnmount() {
    if (!this.props.saved) {
      this.props.clearTheme();
    }
  }

  getNextState = ({ primary, secondary, hue }) => ({
    primaryColor: `$md-${primary}-500`,
    secondaryColor: `$md-${secondary}-a-${hue}`,
  });

  handleChange = (e) => {
    const { updateTheme } = this.props;
    const { value, checked, type } = e.target;
    let { id } = e.target;
    if (id === 'save-theme') {
      id = 'saved';
    }

    updateTheme(id, type === 'checkbox' ? checked : value);
  };

  handleSelectChange = (value, items, e, field) => {
    this.handleChange({ target: field });
  };

  render() {
    const { primaryColor, secondaryColor } = this.state;
    const { light, hue } = this.props;
    const { href, updateTheme, clearTheme, ...props } = this.props; // eslint-disable-line no-unused-vars
    let howToUse = `
#### Using with Sass

\`\`\`scss
@import '~react-md/src/scss/react-md';

$md-light-theme: ${light};${light ? ' // optional for light theme' : ''}
$md-primary-color: ${primaryColor};
$md-secondary-color: ${secondaryColor};

@include react-md-everything;

// Or for a subsection
@include react-md-theme-everything(${primaryColor}, ${secondaryColor}, $md-light-theme, 'custom-theme');
\`\`\`

#### Pre-compiled Themes
    `;

    if (hue === 400 && light) {
      howToUse = `${howToUse}
##### SCSS Import
\`\`\`scss
@import '~react-md/dist/${href}';
\`\`\`

##### CDN
\`\`\`html
<link rel="stylesheet" href="//unpkg.com/react-md/dist/${href}">
\`\`\`
`;
    } else {
      howToUse = `${howToUse}
This current theme is unavailable as a precomiled package. Only accents of \`400\` and the \`light-theme\` have been
precompiled.
`;
    }
    return (
      <div className="md-grid">
        <Helmet title="Theme Builder" />
        <Configuration {...props} onChange={this.handleChange} onSelectChange={this.handleSelectChange} />
        <section className="md-cell md-cell--8 md-cell--6-desktop">
          <Markdown markdown={ABOUT_THEME_BUILDER} className="md-text-container" />
          <Preview />
        </section>
        <Markdown markdown={howToUse} className="md-cell md-cell--12" />
      </div>
    );
  }
}

export default connectAdvanced((dispatch) => {
  let result;
  const actions = bindActionCreators({ updateTheme, clearTheme }, dispatch);
  return (state) => {
    const nextResult = { ...state.theme, ...actions };
    if (!shallowEqual(result, nextResult)) {
      result = nextResult;
    }

    return result;
  };
})(PureThemeBuilder);
