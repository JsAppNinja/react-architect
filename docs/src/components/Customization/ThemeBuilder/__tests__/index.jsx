/* eslint-env jest */
import React from 'react';
import { renderReduxRouterSnapshot } from 'utils/testing';

import ThemeBuilder from '../';

describe('ThemeBuilder', () => {
  it('should render correctly', () => {
    const tree = renderReduxRouterSnapshot(<ThemeBuilder />);
    expect(tree).toMatchSnapshot();
  });
});
