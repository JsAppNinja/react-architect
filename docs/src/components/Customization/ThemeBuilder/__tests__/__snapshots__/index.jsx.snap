// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeBuilder should render correctly 1`] = `
<div
  class="md-grid"
>
  <form
    class="md-cell md-cell--8 md-cell--6-desktop"
    id="theme-configuration-form"
    name="theme-configuration"
  >
    <h3
      class="md-title"
    >
      Theme Configuration
    </h3>
    <div
      class="md-selection-control-container"
    >
      <input
        aria-hidden="true"
        checked=""
        class="md-selection-control-input"
        id="light"
        name="theme-configuration"
        type="checkbox"
      />
      <label
        class="md-selection-control-label md-pointer--hover md-text"
        for="light"
      >
        <div
          aria-checked="true"
          aria-pressed="false"
          class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--theme-secondary"
          role="checkbox"
          tabindex="0"
        >
          <div
            class="md-ink-container"
          />
          <i
            class="md-icon material-icons md-text--inherit"
          >
            check_box
          </i>
        </div>
        <span>
          Light theme
        </span>
      </label>
    </div>
    <div
      class="md-selection-control-container"
    >
      <input
        aria-hidden="true"
        class="md-selection-control-input"
        disabled=""
        id="save-theme"
        name="theme-configuration"
        type="checkbox"
      />
      <label
        class="md-selection-control-label md-text--disabled"
        for="save-theme"
      >
        <div
          aria-checked="false"
          aria-pressed="false"
          class="md-fake-btn md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--disabled"
          disabled=""
          role="checkbox"
        >
          <i
            class="md-icon material-icons md-text--inherit"
          >
            check_box_outline_blank
          </i>
        </div>
        <span>
          Save for future visits
        </span>
      </label>
    </div>
    <p
      class="theme-configuration__message"
    >
      <i>
        When the save for future visits checkbox is checked, cookies will be created to store your theme for later visits and other pages. If you do not check this checkbox, the default theme will be applied when you leave this page.
      </i>
    </p>
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="md-layover md-inline-block md-menu-container md-menu--select-field md-cell"
      id="primary-menu"
    >
      <span
        aria-hidden="true"
      />
      <div
        class="md-select-field__toggle"
      >
        <label
          class="md-floating-label md-floating-label--floating md-text--secondary"
          for="primary"
        >
          Primary color
        </label>
        <div
          aria-pressed="false"
          class="md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text"
          id="primary-toggle"
          role="listbox"
          tabindex="0"
        >
          <div
            class="md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field"
          >
            <span
              class="md-icon-text"
            >
              light-blue
            </span>
            <i
              class="md-icon material-icons"
            >
              arrow_drop_down
            </i>
          </div>
          <hr
            class="md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field"
          />
          <input
            id="primary"
            type="hidden"
            value="light-blue"
          />
        </div>
      </div>
    </div>
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="md-layover md-inline-block md-menu-container md-menu--select-field md-cell"
      id="secondary-menu"
    >
      <span
        aria-hidden="true"
      />
      <div
        class="md-select-field__toggle"
      >
        <label
          class="md-floating-label md-floating-label--floating md-text--secondary"
          for="secondary"
        >
          Secondary color
        </label>
        <div
          aria-pressed="false"
          class="md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text"
          id="secondary-toggle"
          role="listbox"
          tabindex="0"
        >
          <div
            class="md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field"
          >
            <span
              class="md-icon-text"
            >
              deep-orange
            </span>
            <i
              class="md-icon material-icons"
            >
              arrow_drop_down
            </i>
          </div>
          <hr
            class="md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field"
          />
          <input
            id="secondary"
            type="hidden"
            value="deep-orange"
          />
        </div>
      </div>
    </div>
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="md-layover md-inline-block md-menu-container md-menu--select-field md-cell"
      id="hue-menu"
    >
      <span
        aria-hidden="true"
      />
      <div
        class="md-select-field__toggle"
      >
        <label
          class="md-floating-label md-floating-label--floating md-text--secondary"
          for="hue"
        >
          Secondary hue
        </label>
        <div
          aria-pressed="false"
          class="md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text"
          id="hue-toggle"
          role="listbox"
          tabindex="0"
        >
          <div
            class="md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field"
          >
            <span
              class="md-icon-text"
            >
              200
            </span>
            <i
              class="md-icon material-icons"
            >
              arrow_drop_down
            </i>
          </div>
          <hr
            class="md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field"
          />
          <input
            id="hue"
            type="hidden"
            value="200"
          />
        </div>
      </div>
    </div>
  </form>
  <section
    class="md-cell md-cell--8 md-cell--6-desktop"
  >
    <section
      class="markdown-container md-text-container"
    >
      <h3
        id="custom-css-theme-builder"
      >
        Custom CSS Theme Builder
      </h3>
      

      <p>
        Select a primary color, a secondary color, the secondary color's hue, and optionally toggle the light theme
to view a specific theme. When you have selected colors you like, either reference 
        <a
          class="link"
          href="#using-with-sass"
        >
          Using with Sass
        </a>
        
or 
        <a
          class="link"
          href="#pre-compiled-themes"
        >
          pre-compiled themes
        </a>
        . Not all themes will already be compiled and hosted on 
        <code>
          unpkg
        </code>
        .
      </p>
      

    </section>
    <section
      class="md-background--card theme-preview"
    >
      <header
        class="md-paper md-paper--2 md-toolbar md-background--primary md-toolbar--text-white md-toolbar--fixed"
        id="theme-builder-preview-toolbar"
      >
        <button
          class="md-btn md-btn--icon md-pointer--hover md-inline-block md-btn--toolbar md-toolbar--action-left"
          id="theme-builder-preview-drawer-toggle"
          type="button"
        >
          <div
            class="md-ink-container"
          />
          <i
            class="md-icon material-icons md-text--inherit"
          >
            menu
          </i>
        </button>
        <h2
          class="md-title md-title--toolbar"
          id="theme-builder-preview-toolbar-title"
        >
          Theme Preview
        </h2>
      </header>
      <div
        class="md-grid md-toolbar-relative"
      >
        <h2
          class="md-cell md-cell--12 md-display-1"
        >
          Look at this
        </h2>
        <button
          class="md-btn md-btn--raised md-btn--text md-pointer--hover md-background--primary md-background--primary-hover md-inline-block theme-preview__btn"
          id="theme-builder-preview-btn"
          type="button"
        >
          <div
            class="md-ink-container"
          />
          Button
        </button>
      </div>
      <button
        class="md-btn md-btn--icon md-btn--floating md-btn--fixed md-btn--fixed-br md-pointer--hover md-paper md-paper--2 md-background--secondary md-background--secondary-hover md-inline-block"
        id="theme-builder-preview-floating-btn"
        type="button"
      >
        <div
          class="md-ink-container"
        />
        <i
          class="md-icon material-icons md-text--inherit"
        >
          email
        </i>
      </button>
    </section>
  </section>
  <div
    class="md-cell md-cell--12 markdown-container"
  >
    <h4
      id="using-with-sass"
    >
      Using with Sass
    </h4>
    

    <pre>
      <code
        class="lang-scss"
      >
        @import '~react-md/src/scss/react-md';

$md-light-theme: true; // optional for light theme
$md-primary-color: $md-light-blue-500;
$md-secondary-color: $md-deep-orange-a-200;

@include react-md-everything;

// Or for a subsection
@include react-md-theme-everything($md-light-blue-500, $md-deep-orange-a-200, $md-light-theme, 'custom-theme');

      </code>
    </pre>
    

    <h4
      id="pre-compiled-themes"
    >
      Pre-compiled Themes
    </h4>
    

    <p>
      This current theme is unavailable as a precomiled package. Only accents of 
      <code>
        400
      </code>
       and the 
      <code>
        light-theme
      </code>
       have been
precompiled.
    </p>
    

  </div>
</div>
`;
